src/main/java/com/exasol/adapter/document/documentfetcher/files/AbstractLocalFileLoader.java

```diff
@@ -0,0 +1,84 @@
+package com.exasol.adapter.document.documentfetcher.files;
+
+import java.io.*;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+/**
+ * Abstract basis for {@link FileLoader}s local files.
+ * 
+ * @implNote this class is only used by the {@link BucketfsFileLoader}. It was introduced to support unit testing.
+ */
+abstract class AbstractLocalFileLoader implements FileLoader {
+    public static final Pattern NON_GLOB_PATH_PATTERN = Pattern.compile("([^\\*\\?\\[]*\\/).*");
+    private final String path;
+    private final SegmentMatcher segmentMatcher;
+
+    /**
+     * Create a new instance of {@link AbstractLocalFileLoader}.
+     * 
+     * @param baseDirectory      base directory configured in the CONNECTION
+     * @param filePattern        GLOB pattern for the file set to load
+     * @param segmentDescription files to load
+     */
+    public AbstractLocalFileLoader(final String baseDirectory, final String filePattern,
+            final SegmentDescription segmentDescription) {
+        this.path = concatenatePathsInjectionFree(baseDirectory, filePattern);
+        this.segmentMatcher = new SegmentMatcher(segmentDescription);
+    }
+
+    private String concatenatePathsInjectionFree(final String path1, final String path2) {
```

Comment:
Change the parameter paths similar to the constructor parameters: baseDirectory, filePath